modules/programs/neovim.nix

```diff
@@ -53,7 +53,11 @@ let
         (map (x: if x ? plugin && x.optional == true then x.plugin else null)
           cfg.plugins);
     };
-    beforePlugins = "";
+    beforePlugins = ''
+      " generated by Home Manager
+      set nocompatible
```

Comment:
we want home-manager not to generate a file if nothing is configured, this goes against that. Also neovim has  `set nocompatible` by default so no need to pass that.

Reply:
Could you so please provide a better and working solution? This is what I and berbiche were able to make. All the other provided solutions until now lead to nix errors or nvim config duplication, and I do not see why I should write my config trying to work around this, since it's a home-manager bug that should be fixed.

Reply:
Something like this:

``` suggestion
optionalString (cfg.initExtra != "") ''
  " Generated by Home Manager
  ${cfg.initExtra}
''
```

Reply:
ping @arcticlimer